datalad/support/gitrepo.py

```diff
@@ -1917,6 +1907,132 @@ def _run_command_files_split(
                 err += err_
         return out, err
 
+    # Convenience wrappers for one-off git calls that don't require further
+    # processing or error handling.
+
+    def call_git(self, args, files=None,
+                 expect_stderr=False, expect_fail=False,
+                 check_fake_dates=False):
+        """Call git and return standard output.
+
+        Parameters
+        ----------
+        args : list of str
+          Arguments to pass to `git`.
+        files : list of str, optional
+          File arguments to pass to `git`. The advantage of passing these here
+          rather than as part of `args` is that the call will be split into
+          multiple calls to avoid exceeding the maximum command line length.
+        expect_stderr : bool, optional
+          Standard error is expected and should not be elevated above the DEBUG
+          level.
+        expect_fail : bool, optional
+          A non-zero exit is expected and should not be elevated above the
+          DEBUG level.
+        check_fake_dates : boolean, optional
+          DataLad supports faking the dates stored in repository objects. If
+          this call isn't purely for inspection (e.g., it can lead to a new
+          commit), this flag should be set so the dates are overridden.
+
+        Returns
+        -------
+        standard output (str)
+
+        Raises
+        ------
+        GitCommandError if the call exits with a non-zero status.
+        """
+        out, _ = self._git_custom_command(files, ["git"] + args,
+                                          expect_stderr=expect_stderr,
+                                          expect_fail=expect_fail,
+                                          check_fake_dates=check_fake_dates)
+        return out
+
+    def call_git_items(self, args, files=None, expect_stderr=False, sep=None):
+        """Call git, splitting output on `sep`.
+
+        Parameters
+        ----------
+        args : list of str
+          Arguments to pass to `git`.
+        files : list of str, optional
+          File arguments to pass to `git`. The advantage of passing these here
+          rather than as part of `args` is that the call will be split into
+          multiple calls to avoid exceeding the maximum command line length.
+        expect_stderr : bool, optional
+          Standard error is expected and should not be elevated above the DEBUG
+          level.
+        sep : str, optional
+          Split the output by `str.split(sep)` rather than `str.splitlines`.
+
+        Returns
+        -------
+        Generator that yields output items.
```

Comment:
Q (without any hidden messages): Should we do tandems of functions (list and generator) like we did elsewhere, for reasons of consistency? Or should we not?

I think the issue back than was a report on confusion that some processing didn't happen, because the generator wasn't consumed...

However, I am very open to break with this "tradition". I have not found myself switching between these often (or ever), and typically it is the generator that is ultimately the right choice.

Reply:
I think in the present context you aren't missing anything (given the limitations of the present `Runner` class, etc). I was thinking more like: If we ever can or need to handle "long" running processes that yield output line-by-line that could be acted upon, do we want to remain limited by this when installing this new API part now. I have no strong feelings, but it seems now is the time to bring it up.